frontend/app/src/components/settings/accounting/rule/AccountingRuleForm.vue

```diff
@@ -0,0 +1,230 @@
+<script setup lang="ts">
+import { required } from '@vuelidate/validators';
+import { omit } from 'lodash-es';
+import {
+  AccountingMethod,
+  type AccountingRuleEntry,
+  AccountingTreatment
+} from '@/types/settings/accounting';
+import { toMessages } from '@/utils/validation';
+import { ApiValidationError } from '@/types/api/errors';
+
+const props = withDefaults(
+  defineProps<{
+    editableItem?: AccountingRuleEntry | null;
+  }>(),
+  {
+    editableItem: null
+  }
+);
+
+const { editableItem } = toRefs(props);
+
+const getPlaceholderRule = (): AccountingRuleEntry => ({
+  identifier: -1,
+  eventType: '',
+  eventSubtype: '',
+  counterparty: '',
+  taxable: false,
+  countEntireAmountSpend: false,
+  countCostBasisPnl: false,
+  method: AccountingMethod.SPEND,
+  accountingTreatment: AccountingTreatment.SWAP
+});
+
+const state: Ref<AccountingRuleEntry> = ref(getPlaceholderRule());
+
+const rules = {
+  eventType: { required },
+  eventSubtype: { required },
+  counterparty: { externalServerValidation: () => true },
+  method: { required },
+  accountingTreatment: { required }
+};
+
+const { t } = useI18n();
+
+const { setValidation, setSubmitFunc } = useAccountingRuleForm();
+
+const errorMessages = ref<Record<string, string[] | string>>({});
+
+const v$ = setValidation(rules, state, {
+  $autoDirty: true,
+  $externalResults: errorMessages
+});
+
+onMounted(() => {
+  const editable = get(editableItem);
+  if (editable) {
+    reset(editable);
+  }
+});
+
+watch(editableItem, editableItem => {
+  if (editableItem) {
+    reset(editableItem);
+  }
+});
+
+const reset = (newState?: AccountingRuleEntry) => {
+  set(state, newState ?? getPlaceholderRule());
+};
+
+const { addAccountingRule, editAccountingRule } = useAccountingApi();
+const { setMessage } = useMessageStore();
+
+const save = async () => {
+  const editing = !!get(editableItem);
+  const payload = get(state);
+
+  try {
+    const result = editing
+      ? await editAccountingRule(payload)
+      : await addAccountingRule(omit(payload, 'identifier'));
+
+    if (result) {
+      reset();
+    }
+
+    return result;
+  } catch (e: any) {
+    const errorTitle = editing
+      ? t('accounting_settings.rule.edit_error')
+      : t('accounting_settings.rule.add_error');
+
+    if (e instanceof ApiValidationError) {
+      const messages = e.errors;
+
+      set(errorMessages, messages);
+
+      const keys = Object.keys(messages);
+      const knownKeys = Object.keys(payload);
+      const unknownKeys = keys.filter(key => !knownKeys.includes(key));
```

Comment:
I guess this differs a bit from the `getValidationErrors` method but maybe we could also integrate that in the ApiValidationError class too as an extra method.

Reply:
Sorry that was old implementation, updated to new one that use `getValidationErrors`